core/src/com/unciv/logic/map/mapunit/UnitMovement.kt

```diff
@@ -726,6 +726,9 @@ class UnitMovement(val unit: MapUnit) {
             if (!unitSpecificAllowOcean && unit.cache.cannotEnterOceanTiles) return false
         }
 
+        if (unit.getMatchingUniques(UniqueType.CanTradeWithCityStateForGoldAndInfluence)
```

Comment:
Does HasUnique consider state for conditionals? If so that's what you should be using